src/modules/interactiveFunctions.js

```diff
@@ -0,0 +1,72 @@
+function checkboxValidation(taskList) {
+  const checkboxInputs = document.querySelectorAll('input[type="checkbox"]');
+  let completedTasks = [];
+
+  // Setting local storage completed tasks if exist
+  if (localStorage.getItem('checkboxStatus')) {
+    const checkboxStatus = JSON.parse(localStorage.getItem('checkboxStatus'));
+
+    for (let index = 0; index < checkboxStatus.length; index += 1) {
+      if (checkboxStatus[index] === true) completedTasks.push(index);
+    }
+  }
+  checkboxInputs.forEach((checkboxInput) => {
+    checkboxInput.addEventListener('change', () => {
+      const taskRow = checkboxInput.parentNode.parentNode;
+      const ellipsisIcon = taskRow.querySelector('.fa-ellipsis-v');
+      const trashIcon = taskRow.querySelector('.fa-trash');
+      const taskDescription = taskRow.querySelector('.task-info label');
+
+      const checkboxStatus = [];
+      if (checkboxInput.checked) {
+        // Show trash icon and add task to completed tasks array
+        ellipsisIcon.classList.add('hide');
+        trashIcon.classList.remove('hide');
+        taskDescription.classList.add('completed-task-text');
+        completedTasks.push(Number(taskRow.id));
+
+        // Set checkbox status true
+        taskList[Number(taskRow.id)].completed = true;
+        taskList.forEach((task) => {
+          checkboxStatus.push(task.completed);
+        });
+        localStorage.setItem('checkboxStatus', JSON.stringify(checkboxStatus));
+      } else {
+        // Show ellipsis icon and remove task from completed tasks array
+        ellipsisIcon.classList.remove('hide');
+        trashIcon.classList.add('hide');
+        taskDescription.classList.remove('completed-task-text');
+        completedTasks = completedTasks.filter((id) => Number(taskRow.id) !== id);
+
+        // Set checkbox status false
+        taskList[Number(taskRow.id)].completed = false;
+        taskList.forEach((task) => {
+          checkboxStatus.push(task.completed);
+        });
+        localStorage.setItem('checkboxStatus', JSON.stringify(checkboxStatus));
+      }
+    });
```

Comment:
- [ ] Hello @NicolasAndrehh , You have made a great effort working on your to-do list interactivity, but currently a strange functionality different from the one we are asked to implement in the project requirement. I noticed the checkbox is not updating the completed function. Whenever a list is checked it only changes the checkbox status alone and the completed did not change from `false` to `true`. before the status of the completed changes you will have to add a new to-do to the list then the previously checked list will now change status.
    - Note that the required functionality is that whenever a list is checked, the completed status should change from `false` to `true` automatically.
    
Video of my claim 👇 

[screen-capture (7).webm](https://user-images.githubusercontent.com/78634841/220800482-2444e683-5461-4425-a090-73a25fb6f7a2.webm)

Reply:
Hey, @brainconnect93 thank you so much for your feedback, I really appreciate that you did make that video because it helped me a lot to understand your suggestion. 😄